libraries/kmp-iap/library/src/androidMain/kotlin/io/github/hyochan/kmpiap/openiap/WebhookTransport.android.kt

```
@@ -0,0 +1,267 @@
+                        }
+                    }
+                }
+            } catch (cancellation: CancellationException) {
+                // Coroutine cancellation must propagate so the
+                // collector can tear down — wrapping it in the
+                // generic Throwable catch below would treat the
+                // cancellation as a transient transport error and
+                // re-enter the retry loop. Re-throw before any
+                // back-off / reconnect logic runs.
+                throw cancellation
+            } catch (error: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)